server/src/main/java/com/flipkart/varadhi/web/routes/RouteDefinition.java

```diff
@@ -18,7 +19,8 @@
 
 @Slf4j
 public record RouteDefinition(HttpMethod method, String path, Set<RouteBehaviour> behaviours,
-                              Handler<RoutingContext> handler,
+                              List<Handler<RoutingContext>> preHandlers,
```

Comment:
Why list instead of set ?

Reply:
Set will require "equality" among elements. Handlers dont have that. also list will have defined iteration order (which is same as insertion order)